src/pybind/ceph_argparse.py

```diff
@@ -1200,14 +1200,13 @@ def run(self):
             self.exception = e
 
 
-# time in seconds between each call to t.join() for child thread
-POLL_TIME_INCR = 0.5
-
-
 def run_in_thread(func, *args, **kwargs):
     interrupt = False
     timeout = kwargs.pop('timeout', 0)
-    countdown = timeout
+    if timeout == 0:
+        # python threading module will just get blocked if timeout is `None`,
+        # otherwise it will keep polling until timeout or thread stops.
+        timeout = 2 ** 32
```

Comment:
Won't it be nicer to assign this value to timeout directly rather that assigning 0 and then checking for it?

Reply:
what if the caller passes `timeout = 0`?

Reply:
Yeah, you are right.

These statements can seem confusing since one can get the impression that the default value we want is zero. Maybe we can rewrite as following -

```
    timeout = kwargs.pop('timeout', 2 ** 32)
    if timeout == 0:
        # python threading module will just get blocked if timeout is `None`,
        # otherwise it will keep polling until timeout or thread stops.
        timeout = 2 ** 32
```

However, this might only be my personal opinion. Feel free to merge if you disagree.  :)

Reply:
personally, i don't like the magic number of `2 ** 32`, but there is no simple way to put it in python. so i'd prefer putting it for once instead of twice. and 0 is less magical than `2 ** 32` is. so using it as the default value might be less confusing, IMHO.

Reply:
Okay.